Modules/Scripted/SegmentStatistics/SegmentStatisticsCalculators/LabelmapSegmentStatisticsCalculator.py

```diff
@@ -0,0 +1,94 @@
+    self.defaultKeys = self.keys # calculate all measurements by default
+    #... developer may add extra options to configure other parameters
+
+  def computeStatistics(self, segmentID):
+    import vtkSegmentationCorePython as vtkSegmentationCore
+    requestedKeys = self.getRequestedKeys()
+
+    segmentationNode = slicer.mrmlScene.GetNodeByID(self.getParameterNode().GetParameter("Segmentation"))
+
+    if len(requestedKeys)==0:
+      return {}
+
+    containsLabelmapRepresentation = segmentationNode.GetSegmentation().ContainsRepresentation(
+      vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName())
+    if not containsLabelmapRepresentation:
+      return {}
+
+    segment = segmentationNode.GetSegmentation().GetSegment(segmentID)
+    segBinaryLabelName = vtkSegmentationCore.vtkSegmentationConverter.GetSegmentationBinaryLabelmapRepresentationName()
+    segmentLabelmap = segment.GetRepresentation(segBinaryLabelName)
+
+    # We need to know exactly the value of the segment voxels, apply threshold to make force the selected label value
+    labelValue = 1
+    backgroundValue = 0
+    thresh = vtk.vtkImageThreshold()
+    thresh.SetInputData(segmentLabelmap)
+    thresh.ThresholdByLower(0)
+    thresh.SetInValue(backgroundValue)
+    thresh.SetOutValue(labelValue)
+    thresh.SetOutputScalarType(vtk.VTK_UNSIGNED_CHAR)
+    thresh.Update()
+
+    #  Use binary labelmap as a stencil
+    stencil = vtk.vtkImageToImageStencil()
+    stencil.SetInputData(thresh.GetOutput())
+    stencil.ThresholdByUpper(labelValue)
+    stencil.Update()
+
+    stat = vtk.vtkImageAccumulate()
+    stat.SetInputData(thresh.GetOutput())
+    stat.SetStencilData(stencil.GetOutput())
+    stat.Update()
+
+    # Add data to statistics list
+    cubicMMPerVoxel = reduce(lambda x,y: x*y, segmentLabelmap.GetSpacing())
+    ccPerCubicMM = 0.001
+    stats = {}
+    if "Labelmap.voxel_count" in requestedKeys:
+      stats["Labelmap.voxel_count"] = stat.GetVoxelCount()
+    if "Labelmap.volume_mm3" in requestedKeys:
+      stats["Labelmap.volume_mm3"] = stat.GetVoxelCount() * cubicMMPerVoxel
+    if "Labelmap.volume_cc" in requestedKeys:
+      stats["Labelmap.volume_cc"] = stat.GetVoxelCount() * cubicMMPerVoxel * ccPerCubicMM
+    return stats
+
+  def getMeasurementInfo(self, key):
+    """Get information (name, description, units, ...) about the measurement for the given key"""
+    info = {}
+
+    # @fedorov could not find any suitable code.
+    # DCM has "Number of needles" etc., so probably "Number of voxels"
+    # should be added too. Need to discuss with @dclunie. For now, a
+    # QIICR private scheme placeholder.
+
+    info["Labelmap.voxel_count"] = \
+      self.generateMeasurementInfo(name="Voxel count", description="Number of voxels", units="voxels",
+                                   quantityCode=self.initCodedEntry("nvoxels", "99QIICR", "Number of voxels", True),
```

Comment:
CodeValue is SH VR, so it can be at most 16 characters (see http://dicom.nema.org/dicom/2013/output/chtml/part05/sect_6.2.html). Inevitably, we will have constraints about how expressive we could be. If we choose a convention like you suggest, we eventually will run into a problem when the approach will hit the 16 character limit, and then the question will become how to abbreviate. I personally think it doesn't matter,  since CodeValue is not expected to be readable. As long as it is unique, it is not a big deal. 

The easiest and least contentious I guess is to just assign a number, and have a central location for contributing new terms to eliminate the possibility of clashes.